packages/peregrine/lib/context/user.js

```diff
@@ -23,6 +24,22 @@ const UserContextProvider = props => {
         userState
     ]);
 
+    useEffect(() => {
```

Comment:
How often does this run? And does it always run in the necessary circumstance?

Reply:
It only runs once on refresh and it runs on time as needed by the logic to work.

Reply:
I like this idea except for the refresh part, since that may not happen in an installed PWA. Can the token expire while the user is browsing around? If so, what happens then?

Reply:
I actually hope this can _not_ happen since that would be a terrible user experience to be logged out in the middle of shopping 🙃

Reply:
Well it turns out that's exactly how it works

Reply:
The use case we were trying to solve, includes a refresh or reopening of a closed tab. If the token expires mid-way while browsing, don't we have a refresh token or some mechanism to handle it?

Reply:
I tried the same without refresh and everything worked as expected. The app logged the user out and cleared cache..

Reply:
Ah yes, there is something else for that. Nice!